openquake/risklib/riskinput.py

```diff
@@ -0,0 +1,382 @@
+        ...  (0, 'AkkarBommer2010'): numpy.array([0.05, 0.03, 0.04])}])
+        {0: [array([ 0.05,  0.03,  0.04])], 1: [array([ 0.06,  0.02,  0.09])]}
+        """
+        keys = sorted(dicts[0])
+        values_by_rlz = {i: [] for i in range(len(keys))}
+        for dic in map(self.combine, dicts):
+            if dic:
+                for i in range(len(keys)):
+                    values_by_rlz[i].append(dic[i])
+        return values_by_rlz
+
+    def __iter__(self):
+        return self.rlzs_assoc.iterkeys()
+
+    def __getitem__(self, key):
+        return self.rlzs_assoc[key]
+
+    def __len__(self):
+        return len(self.rlzs_assoc)
+
+
+class RiskModel(collections.Mapping):
+    """
+    A container (imt, taxonomy) -> workflow
+    """
+    def __init__(self, workflows, damage_states=None):
+        self.damage_states = damage_states  # not None for damage calculations
+        self._workflows = workflows
+
+    def get_loss_types(self):
+        """
+        :returns: a sorted list with all the loss_types contained in the model
+        """
+        return sorted(set(sum([w.loss_types for w in self.values()], [])))
+
+    def get_taxonomies(self, imt=None):
+        """
+        :returns: the set of taxonomies which are part of the RiskModel
+        """
+        if imt is None:
+            return set(taxonomy for imt, taxonomy in self)
+        return set(taxonomy for imt_str, taxonomy in self if imt_str == imt)
+
+    def get_imts(self, taxonomy=None):
+        if taxonomy is None:
+            return set(imt for imt, taxonomy in self)
+        return set(imt for imt, taxo in self if taxo == taxonomy)
+
+    def get_imt_taxonomies(self):
+        """
+        For each IMT in the risk model, yield pairs (imt, taxonomies)
+        with the taxonomies associated to the IMT. For fragility functions,
+        there is a single taxonomy for each IMT.
+        """
+        by_imt = operator.itemgetter(0)
+        by_taxo = operator.itemgetter(1)
+        for imt, group in itertools.groupby(sorted(self), key=by_imt):
+            yield imt, map(by_taxo, group)
+
+    def __getitem__(self, imt_taxo):
+        return self._workflows[imt_taxo]
+
+    def __iter__(self):
+        return iter(sorted(self._workflows))
+
+    def __len__(self):
+        return len(self._workflows)
+
+    def build_input(self, imt, hazards_by_site, assets_by_site):
+        """
+        :param imt: an Intensity Measure Type
+        :param hazards_by_site: an array of hazards per each site
+        :param assets_by_site: an array of assets per each site
+        :returns: a :class:`RiskInput` instance
+        """
+        imt_taxonomies = [(imt, self.get_taxonomies(imt))]
+        return RiskInput(imt_taxonomies, hazards_by_site, assets_by_site)
+
+    def build_input_from_ruptures(self, sitecol, assets_by_site, ses_ruptures,
+                                  gsims, trunc_level, correl_model, eps_dict):
+        """
```

Comment:
You forgot to re-write the docstring for this method (currently it is the same as in the previous method)